packages/flutter_tools/test/general.shard/asset_test.dart

```diff
@@ -131,13 +131,187 @@ dependencies:
+
+environment:
+  sdk: ^3.7.0-0
+
+flutter:
+  uses-material-design: true
+  fonts:
+  - family: test_font
+    fonts:
+      - asset: test_font_file
+''');
+          fileSystem.file(fileSystem.path.join('font', 'test_font_file'))
+            ..createSync(recursive: true)
+            ..writeAsStringSync('This is a fake font.');
+
+          fileSystem.file(fileSystem.path.join('dev_dependency', 'pubspec.yaml'))
+            ..createSync(recursive: true)
+            ..writeAsStringSync(r'''
+name: dev_dependency
+description: Another test project that contains a font.
+
+environment:
+  sdk: ^3.7.0-0
+
+flutter:
+  uses-material-design: true
+  fonts:
+  - family: dev_dependency_test_font
+    fonts:
+      - asset: dev_dependency_test_font_file
+''');
+          fileSystem.file(fileSystem.path.join('dev_dependency', 'dev_dependency_test_font_file'))
+            ..createSync(recursive: true)
+            ..writeAsStringSync('This is a fake font.');
+
+          fileSystem.file(fileSystem.path.join('dev_dependency', 'pubspec.yaml'))
+            ..createSync(recursive: true)
+            ..writeAsStringSync(r'''
+name: dev_dependency
+description: Another test project that contains a font.
+
+environment:
+  sdk: ^3.7.0-0
+
+flutter:
+  uses-material-design: true
+  fonts:
+  - family: dev_dependency_test_font
+    fonts:
+      - asset: dev_dependency_test_font_file
+''');
+          fileSystem.file(fileSystem.path.join('dev_dependency', 'dev_dependency_test_font_file'))
+            ..createSync(recursive: true)
+            ..writeAsStringSync('This is a fake font.');
+
+          fileSystem.file(packageConfigPath)
+            ..createSync(recursive: true)
+            ..writeAsStringSync(r'''
+{
+  "configVersion": 2,
+  "packages": [
+    {
+      "name": "font",
+      "rootUri": "../font",
+      "packageUri": "lib/",
+      "languageVersion": "3.7"
+    },
+    {
+      "name": "main",
+      "rootUri": "../main",
+      "packageUri": "lib/",
+      "languageVersion": "3.7"
+    },
+    {
+      "name": "workspace_root",
+      "rootUri": "../",
+      "packageUri": "lib/",
+      "languageVersion": "3.7"
+    }
+  ],
+  "generated": "2024-01-08T19:39:02.396620Z",
+  "generator": "pub",
+  "generatorVersion": "3.3.0-276.0.dev"
+}
+''');
+          fileSystem.file(manifestPath)
+            ..createSync(recursive: true)
+            ..writeAsStringSync(r'''
+name: main
+description: A test project that has a package with a font as a dependency.
+
+environment:
+  sdk: ^3.7.0-0
+
+resolution: workspace
+
+dependencies:
+  font:
+    path: ../font
+
+dev_dependencies:
+  dev_dependency:
+    path: ../dev_dependency
+''');
+
+          fileSystem.file('pubspec.yaml')
+            ..createSync(recursive: true)
+            ..writeAsStringSync(r'''
+name: workspace_root
+description: A pubspec that defines a workspace root, but also a font.
+
+environment:
+  sdk: ^3.7.0-0
+
+workspace:
+  - main/
+
+flutter:
+  uses-material-design: true
```

Comment:
Is this true while the main app is false intentionally?

Reply:
I don't remember - probably a mistake - I'll turn it to true in the main app also.